test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go

```diff
@@ -254,6 +374,77 @@ func objectHPA(namespace string, metricTarget int64) *as.HorizontalPodAutoscaler
 	}
 }
 
+type externalMetricTarget struct {
+	value     int64
+	isAverage bool
+}
+
+func externalHPA(namespace string, metricTargets map[string]externalMetricTarget) *as.HorizontalPodAutoscaler {
+	var minReplicas int32 = 1
+	metricSpecs := []as.MetricSpec{}
+	selector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{"resource.type": "gke_container"},
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "resource.labels.namespace_id",
+				Operator: metav1.LabelSelectorOpIn,
+				// TODO(bskiba): change default to real namespace name it is available
+				// from Stackdriver.
+				Values: []string{"default", "dummy"},
+			},
+			{
+				Key:      "resource.labels.pod_id",
+				Operator: metav1.LabelSelectorOpExists,
+				Values:   []string{},
+			},
+			{
+				Key:      "resource.labels.zone",
```

Comment:
Can you add a comment explaining why we add "zone" and "cluster_name" selectors?

Reply:
Removed these, as they are not really testing anything and only confuse.